snmalloc-sys/build.rs

```diff
@@ -49,15 +50,17 @@ fn main() {
             cfg = cfg.define("ANDROID_ABI", "armeabi-v7a");
         }
     }
+	//I've had strange issues with config route so prefer this method
+    let target_env = env::var("CARGO_CFG_TARGET_ENV").unwrap();
 
-    if cfg!(all(windows, target_env = "msvc")) {
+    if target_env == "msvc" {
         cfg = cfg.define("CMAKE_CXX_FLAGS_RELEASE", "/O2 /Ob2 /DNDEBUG /EHsc");
         cfg = cfg.define("CMAKE_C_FLAGS_RELEASE", "/O2 /Ob2 /DNDEBUG /EHsc");
-    }
-
-    if cfg!(all(windows, target_env = "gnu")) {
+    } else if target_env == "gnu" {
         cfg = cfg.define("CMAKE_SH", "CMAKE_SH-NOTFOUND");
-    }
+        cfg = cfg.define("CMAKE_CXX_FLAGS_RELEASE", "-ffunction-sections -fdata-sections -m64 -O3 -fpic");
```

Comment:
Great. But why do you leave sections flags set on default? Do you need them for PGO or debug?

Reply:
Thanks for pointing that out, I forgot the linker flags:
        cfg = cfg.define("CMAKE_LINKER_FLAGS_RELEASE", "-gc-sections");
I can add this or we can remove all together its not needed for fpic, it's most used to optimize individual functions, which makes since when you are not creating a whole executable but only pulling sections of the code from it, quick calls to malloc for example at least in my interpretation. gc-sections removes unused functions at the linking process and can help make a smaller statically linked executable.

https://gcc.gnu.org/onlinedocs/gcc/Optimize-Options.html#index-fdata-sections